File: cross-application-feature-development-management/Directories/Feature/AutomationsDirectory/OperationsDirectory/OperationsDirectory.cs
namespace cross_application_feature_development_management.Directories.Feature.AutomationsDirectory.OperationsDirectory
{
    public class OperationsDirectory(
        AutomationsDirectory automationsDirectory,
        Directories directories
        )
    {
        public void Create()
        {
            var path = GetPath();
            Directory.CreateDirectory(path);
        }
        public string GetPath()
        {
            var directory = automationsDirectory.GetPath();
            var operationsDirectory = Path.Combine(directory, "operations");
            return operationsDirectory;
        }

        public void ReplaceFileNamesWithPaths(string giversPath)
        {
            var pathToTarget = GetPath();
            foreach (var filePath in Directory.EnumerateFiles(pathToTarget))
            {
                var fileName = Path.GetFileNameWithoutExtension(filePath);
                var giverFileName = $"{fileName}.ps1";
                var giverPath = Path.Combine(giversPath, giverFileName);
                directories.ReplaceFileNameWithPath(filePath, giverPath);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/Automations/Operations/OperationsDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.Automations.Operations

open System.IO
open cafdemwimu.console.Directories
open cafdemwimu.console.Directories.Hosting.Feature.Automations

type OperationsDirectory(automationsDirectory: AutomationsDirectory) =
    member this.Create() =
        let path = this.GetPath()
        Directory.CreateDirectory(path) |> ignore

    member _.GetPath() =
        let directory = automationsDirectory.GetPath()
        let operationsDirectory = Path.Combine(directory, "operations")
        operationsDirectory

    member this.ReplaceFileNamesWithPaths() =
        let giversPath = Path.Combine(automationsDirectory.GetPath(), "commands")
        let pathToTarget = this.GetPath()
        for filePath in Directory.EnumerateFiles(pathToTarget) do
            let fileName = Path.GetFileNameWithoutExtension(filePath)
            let giverFileName = $"{fileName}.ps1"
            let giverPath = Path.Combine(giversPath, giverFileName)
            DirectoryServices.ReplaceFileNameWithPath(filePath, giverPath)
